### ISSUE TITLE
Anchor tags in megamenu cause menus to close

### ISSUE THREAD

**Issue by TylerOrtiz** [vadikom/smartmenus#27]
In the smartmenus prototype, I have noticed that clicking on anchor tags causes the smartmenu to close. This causes an issue for me using bootstrap accordions and panels inside of the megamenu because in the all encompassing docClick method there is a check to see if the target clicked is an anchor tag (which bootstrap uses for tons of things by default). 

Calling smartmenus plugin with hideOnClick = false did not seem to help the issue, I could not hide the menu no matter what using the 'menuHideAll' api or global $.SmartMenus.hideAll() method:
```javascript
$('#main-menu').smartmenus({
            hideOnClick: false
});
```
The condition **$(e.target).is('a')** is what I'm referring to in particular:
```javascript
docClick: function(e) {
				if (this.isTouchScrolling) {
					this.isTouchScrolling = false;
					return;
				}
				// hide on any click outside the menu or on a menu link
				if (this.visibleSubMenus.length && !$.contains(this.$root[0], e.target) || $(e.target).is('a')) {
					this.menuHideAll();
				}
			},
```

**Comment by vadikom** [comment]
Sorry for the late reply! It's been very hectic times here lately. :(

I can see the issue you are facing. To be honest, I am hesitant if it's a good idea to change the behavior for links inside mega dropdown sub menus since in many cases the way it currently works would still be the expected behavior (i.e. hide the sub menu on click even for links in mega dropdowns). So I would suggest you to use a simple additional fix:

```javascript
// do not hide the mega dropdown sub menu if a link inside it is clicked
$(document).on('click', 'ul.mega-menu a', function(e) {
	e.stopPropagation();
});
```

It will not hurt your configuration if this behavior is changed in the script core in the future and you decide to upgrade.

**Comment by TylerOrtiz** [comment]
I tried stopping propagation but it disabled bootstrap's elements relying on clicks too (naturally). I did change the code in the plugin to use an and operator (&&) from an or (||) and this seemed to do the trick nicely. I am aware changing plugins is a bad idea but in the interest of time this seemed to solve everything nicely and I documented the change in our own version of the plugin.

If you're interested, I just changed:
```javascript
if (this.visibleSubMenus.length && !$.contains(this.$root[0], e.target) || $(e.target).is('a')) {
```
to:
```javascript
if (this.visibleSubMenus.length && !$.contains(this.$root[0], e.target) && $(e.target).is('a')) {
```

I appreciate the help though, thank you vadikom.

**Comment by vadikom** [comment]
I am not sure if you tested with the exact code I posted, but I believe it should work correctly with all Bootstrap JS components. I tested putting an accordion in a mega dropdown:

http://getbootstrap.com/javascript/#collapse-example-accordion

and the clicks work just fine for the accordion and do not hide the sub menu.

The change you've done means that the sub menus will be hidden only when a link on your page that is not contained in your menu tree is clicked.

Anyway, thanks for your input! Cheers!
